11.1 - all 39 specs

syrifgit opened the pull request:
# 11.1 Omnibus Bill - Fully conflict resolved with the destination branch
> [!TIP]
> All itemized patch notes can be seen in the individual class issues/tickets (linked below)
### List of Blizz 11.1 notes addressed by this PR
- [x] December
- [x] Jan 15
- [x] Jan 23 (pretty minor)
- Jan 28 - NOPE. I am doing these after this PR to keep things organized and reduce mistakes. Brain can only hold so many changes at a time. After this is reviewed, fixed if applicable, and merged, I will start a new PR with this branch as the base for the Jan 28th patch notes.
## Classes Contained In This PR
> [!NOTE]  
> All patch notes are organized into per-class checklists, you can see the full thing by opening the linked issues. Any non-11.1 changes I made are also detailed in those issues (such as model accuracy updates, interaction corrections, etc)
- [x] #4294
  - [x] Proper documentation/notes written
  - [x] Fully tested in-game on PTR
- [x] #4293
  - [x] Proper documentation/notes written
  - [ ] Fully tested in-game on PTR
- [x] #4295
  - [x]  Proper documentation/notes written
  - [x] Fully tested in-game on PTR
- [x] #4301
  - [x] Proper documentation/notes written
  - [x] Fully tested in-game on PTR
- [x] #4303
  - [x] Proper documentation/notes written
  - [x] Fully tested in-game on PTR
- [x] #4300
  - [x] Proper documentation/notes written
  - [ ] Fully tested in-game on PTR
- [x] #4297
  - [x] Proper documentation/notes written
  - [x] Fully tested in-game on PTR
- [x] #4296
  - [x] Proper documentation/notes written
  - [x] Fully tested in-game on PTR
- [x] #4302
  - [x] Proper documentation/notes written
  - [ ] Fully tested in-game on PTR
  - [ ] Final remaining item: Surging Totem - Gains a relocation spell to summon nearby without triggering the global cooldown (6-second cooldown).
- [x] #4292
  - [x] Proper documentation/notes written
  - [ ] Fully tested in-game on PTR
  - [ ] Only remaining item: Ice Lances empowered by Fingers of Frost no longer consume Winter’s Chill.
- [x] #4299
  - [x] Proper documentation/notes written
  - [ ] Fully tested in-game on PTR
  - [ ] Just waiting for Hpal talents to hit PTR
- [x] #4298
  - [x] Proper documentation/notes written
  - [ ] Fully tested in-game on PTR
## Extra Notes / Points of Review
> [!IMPORTANT]  
> These are things I am not 100% sure about my implementation of
### Demon Hunter
- My demonsurge rework
- Removal of the last_meta / last_eyebeam stuff
- My implementation for Vengeance sigil CDR using the table
### Evoker
- My timeToOverride stuff /w Pyre and the State.lua change. Is there a better way to do this? The in-game results are great with this method but it might be dumb.
- Using the `runHandler()` to control a bunch of effects. It seems smart but is it? Am I missing anything?
- Aug empowerment function .. upheaval? action? this_action?
- Dragonrage / deep breath loop to run a spell handler 3 times (it does actually trigger the relevent effects 3x sequentially) .. is there a better way to do that  or is this fine?
### Hunter
- Marksmanship
  - My usage / definitions of `max_targets`, `streamlineCostMultiplier`, `streamlineCastMultiplier`, volley `tick()` function. Did I do a big dumb or should it work as written to reference these expressions more dynamically (and reduce repeated code)? Do I need to define anything elsewhere for the volley ticks?
- BM/Surv
  - Double check my `HowlOfThePackLeaderHandler()` implementation? The buff cycling seemed to work in testing but not 100% sure, first time I have really done something like that.
### Windwalker
- The empowered ability implementation
### Frost Mage
- `brainFreeze()` implementation
### Disc priest
- my bolts thing for penance, similar idea to my Marksmanship Rapid Fire stuff

syrifgit committed:
Support new Marksmanship Lust

syrifgit committed:
Ability timeToReady() overeride

Allows overriding of timeToReady() in cases where we need to manually intervene. Example can be seen in Devastation Evoker's Pyre ability, where this override fixes the rotation and the UI components. Issue is caused by Blizzard registering the ability as on cooldown in relation to resource regeneration (such as essence), even though the addon is fully aware that resource will become available (due to a guaranteed proc) on the next spellcast.

syrifgit committed:
11.1 Hunter specs

syrifgit committed:
11.1 Evoker Specs

syrifgit committed:
Clarifying Comments for similarly named fields

cast vs cost clarification comments. Future me will thank me.

syrifgit committed:
Merge PRs

Merge part 2 into big PR

syrifgit committed:
Merge everything else

squoosh. Except Windwalker, need to redo the ordered elements thing first.

syrifgit committed:
Update PriestDiscipline.lua

syrifgit committed:
/hek set covenants

RussellAnas commented:
Here I was thinking of asking about what the plans are for 11.1 and stumbled upon this thread. Just want to drop in and say, you guys are amazing for already working on upcoming patches while also making sure the current addon works great.

Thank you team!

syrifgit committed:
Windwalker 11.1

Did Ordered Elements the "long way" for now. Structure for a spec-wide spend discount will go on a rainy day list.

syrifgit commented:
Windwalker added

syrifgit committed:
Merge branch 'eleven-one-oh' into syrif-eleven-one-oh

Hekili commented on TheWarWithin/DemonHunterHavoc.lua:
```diff
@@ -550,15 +577,22 @@ spec:RegisterAuras( {
+    },
+    -- https://www.wowhead.com/spell=1215159
+    -- Inertia Your next Fel Rush or Felblade increases your damage by 18% for 5 sec.
+    inertia_prep_buff = { -- find out what simc is gonna use
```
More likely than not, `inertia_prep_buff` will be `inertia_trigger` and `inertia_damage_buff` will be `inertia`.  Should probably use these names for APL backward compatibility (can use `copy = "inertia_trigger",` and keep using these new names for internal clarity.

syrifgit replied:
I'll add the copies for now for all the momentum/inertia/exergy stuff

Hekili commented on TheWarWithin/DemonHunterHavoc.lua:
```diff
@@ -582,18 +616,18 @@ spec:RegisterAuras( {
+        },--]]
     },
-    momentum = {
+    exergy = {
```
Fair chance this will still/also be referrred to as `momentum` in SimC; should make a `copy` so both names are supported.

Hekili commented on TheWarWithin/DemonHunterHavoc.lua:
```diff
@@ -661,9 +695,6 @@ spec:RegisterAuras( {
         max_stack = 1
     },
     reavers_glaive = {
```
?

syrifgit replied:
The old version with a duration and stacks and an id wasn't a real aura in the first place. So I have this empty one just to use for `buff` / `nobuff`. I could put a 3600 duration and max stack?

Hekili replied:
It probably doesn't matter, the old way isn't hurting anything and the new way isn't hurting anything.

Hekili commented on TheWarWithin/DemonHunterHavoc.lua:
```diff
@@ -1857,10 +1854,15 @@ spec:RegisterAbilities( {
         texture = function() return talent.demonic_intensity.enabled and buff.metamorphosis.up and 135794 or 1344649 end,
 
         handler = function ()
+            if IsSpellKnownOrOverridesKnown(452487) then
```
This looks to be already addressed in `reset_precast` when you apply `demonsurge_consuming_fire` based on the spell overlay?  If we do this here, we are delaying/preventing the addon from reacting to the update.  Do we need to use `IsSpelllKnownOrOverridesKnown` in `reset_precast`?

What does this resolve?

syrifgit replied:
-- Consuming Fire, needed because they aren't split into 2 separate ability registrations

Because the code inside the `if IsSpellKnownOrOverridesKnown(452487) then` shouldn't be run if you're casting the regular version of immolation, but the way it was set up was only a single ability registration. To remove this we would have to construct the consuming fire registration and separate the handler logic between the 2 as appropriate.

Maybe it's redundant with the talent + buff up check?

Hekili replied:
Yeah, applying the buff in `reset_precast` should already address this.

Hekili commented on TheWarWithin/DemonHunterHavoc.lua:
```diff
@@ -1933,19 +1925,24 @@ spec:RegisterAbilities( {
-            if level > 19 then stat.haste = stat.haste + 10 end
+            stat.haste = stat.haste + 10
+            -- Legacy
+            if IsSpellKnownOrOverridesKnown( 317009 ) then
```
`if covenant.venthyr then` ?

syrifgit replied:
Sure, that's more clear (and maybe less expensive?)

Hekili commented on TheWarWithin/DemonHunterVengeance.lua:
```diff
@@ -29,136 +29,167 @@ spec:RegisterResource( Enum.PowerType.Fury, {
         value = 2
     },
+    -- 5 fury every 2 seconds for 8 seconds
+    student_of_suffering= {
```
Missing a space.  :)

Hekili commented on State.lua:
```diff
@@ -7499,6 +7499,19 @@ function state:TimeToReady( action, pool )
 
+    -- Early exit for timeToReadyOverride. This is to fix the fact that some spells show as unavailable in game even though
+    -- the addon knows that it will become available on the next GCD due to a guaranteed buff/proc
+    if ability.timeToReadyOverride then
```
Hmm, I need to know more about this -- normally, we address this with a virtual buff or other modifier in `reset_precast`, so it's one-and-done.

syrifgit replied:
Definitely open to implementing another way, this is just what I came up with that achieved the desired result. Can leave it commented out for now (and its partner line in Devasation Evoker Pyre ability registration) as it's not specific to 11.1, just a general improvement.

Hekili replied:
The abilities should be coded such that the guaranteed proc/buff is modeled and the ability checks readiness based on having the buff in place.

syrifgit replied:
I will see if readyTime can forcibly solve it instead. I feel like I tried that, but maybe I didn't.

Hekili replied:
I missed the specific case for this in the code -- let me know and I'll also take a look.

syrifgit replied:
explanation in discord dev chat

Hekili commented on TheWarWithin/DemonHunterVengeance.lua:
```diff
@@ -1112,41 +1173,10 @@ spec:RegisterAbilities( {
-            if legendary.agony_gaze.enabled and debuff.sinful_brand.up then
-                debuff.sinful_brand.expires = debuff.sinful_brand.expires + 0.75
-            end
+            if talent.demonic.enabled then TriggerDemonic() end
```
This happens on cast, instead of at the end of the channel?

syrifgit replied:
Yep, fully tested in game for both veng and havoc. It used to give 5 (or 6 or whatever) seconds at the end of eye beam, now it gives 7 seconds the instant you start channeling. No idea when that change happened, but it did at some point. DH and hunter are probably the most in-game validated/tested.

The way I set up the logic/rules for extensions between demonic and meta, and which demonsurge things re-apply, is also fully tested and confirmed in-game.

Hekili commented on TheWarWithin/DemonHunterVengeance.lua:
```diff
@@ -1177,43 +1207,17 @@ spec:RegisterAbilities( {
-            -- This is likely repeated per tick but it's not worth the CPU overhead to model each tick.
-            if legendary.agony_gaze.enabled and debuff.sinful_brand.up then
-                debuff.sinful_brand.expires = debuff.sinful_brand.expires + 0.75
+            if talent.demonic_intensity.enabled and buff.demonsurge_fel_Desolation.up then
```
Big `D`

Hekili replied:
Resolved but not committed?  Just making sure we don't overlook the capitalization error.

Hekili commented on TheWarWithin/DemonHunterVengeance.lua:
```diff
@@ -1346,18 +1347,24 @@ spec:RegisterAbilities( {
         end,
 
+        tick = function ()
+            if talent.charred_flesh.enabled and debuff.fiery_brand.up then applyDebuff( "target", debuff.fiery_brand.remains + 0.25 * talent.charred_flesh.rank ) end
```
Let's only check the talent 1x.

```Lua
if talent.charred_flesh.enabled then return false end
    if debuff.fiery_brand.up then ...
    if debuff.sigil_of_Flame.up then ...
end
```

Hekili commented on TheWarWithin/DruidFeral.lua:
```diff
@@ -55,155 +55,175 @@ spec:RegisterResource( Enum.PowerType.Mana )
+    dreadful_bleeding              = {  82117, 391045, 1 }, -- Rip damage increased by 20%.
+    feral_frenzy                   = {  82108, 274837, 1 }, -- Unleash a furious frenzy, clawing your target 5 times for 28,039 Physical damage and an additional 185,420 Bleed damage over 4.8 sec. Awards 5 combo points.
+    frantic_momentum               = {  82115, 391875, 2 }, -- Finishing moves have a 3% chance per combo point spent to grant 10% Haste for 6 sec.
+    incarnation_avatar_of_ashamane = {  82114, 102543, 1 }, -- An improved Cat Form that grants all of your known Berserk effects and lasts 20 sec. You may shapeshift in and out of this improved Cat Form for its duration. During Incarnation: Energy cost of all Cat Form abilities is reduced by 25%, and Prowl can be used once while in combat. Generate 1 combo point every 1.5 sec. Combo point genera ...
```
The spec-specific Incarnation needs to be duplicated as `incarnation` (same applies to each Druid spec).

Hekili commented on TheWarWithin/DruidFeral.lua:
```diff
@@ -55,155 +55,175 @@ spec:RegisterResource( Enum.PowerType.Mana )
-    dreadful_bleeding              = { 82117, 391045, 1 }, -- Rip damage increased by 20%.
-    feral_frenzy                   = { 82108, 274837, 1 }, -- Unleash a furious frenzy, clawing your target 5 times for 26,074 Physical damage and an additional 185,091 Bleed damage over 6 sec. Awards 5 combo points.
-    frantic_momentum               = { 82115, 391875, 2 }, -- Finishing moves have a 3% chance per combo point spent to grant 10% Haste for 6 sec.
-    incarnation                    = { 82114, 102543, 1 }, -- An improved Cat Form that grants all of your known Berserk effects and lasts 20 sec. You may shapeshift in and out of this improved Cat Form for its duration. During Incarnation: Energy cost of all Cat Form abilities is reduced by 20%, and Prowl can be used once while in combat. Generate 1 combo point every 1.5 sec. Combo point generat ...
```
Here's the Incarnation that we lost with the refresh.

Hekili commented on TheWarWithin/DruidFeral.lua:
```diff
@@ -2568,12 +2616,16 @@ spec:RegisterAbilities( {
         interrupt = true,
 
-        form = function () return buff.bear_form.up and "bear_form" or "cat_form" end,
+        form = function () if talent.fluid_form.enabled then
```
The formatting of this function tells me that you hate me.

syrifgit replied:
This may have been a late night one. Is this what you expected?

        form = function () if talent.fluid_form.enabled then return end
            return buff.bear_form.up and "bear_form" or "cat_form"
        end,

Hekili replied:
I mean, yeah, that's fine -- the indentation is fixed.

Typically, if a function is multiple lines, I usually bump the first line down, too.  But I'm not requesting that.

```Lua
    form = function ()
        if talent.fluid_form.enabled then return end
        return buff.bear_form.up and "bear_form" or "cat_form"
    end,
```

Hekili commented on TheWarWithin/DruidGuardian.lua:
```diff
@@ -62,160 +61,178 @@ spec:RegisterResource( Enum.PowerType.Energy )
-    gory_fur                      = { 82132, 200854, 1 }, -- Mangle has a 15% chance to reduce the Rage cost of your next Ironfur by 25%.
-    guardian_of_elune             = { 82140, 155578, 1 }, -- Mangle increases the duration of your next Ironfur by 3 sec, or the healing of your next Frenzied Regeneration by 20%.
-    improved_survival_instincts   = { 82128, 328767, 1 }, -- Survival Instincts now has 2 charges.
-    incarnation                   = { 82136, 102558, 1 }, -- An improved Bear Form that grants the benefits of Berserk, causes Mangle to hit up to 3 targets, and increases maximum health by 30%. Lasts 30 sec. You may freely shapeshift in and out of this improved Bear Form for its duration.
```
Save `incarnation`.

syrifgit replied:
Interesting. I wonder if they were originally manually renamed? If not I can take a peek at the skeleton generator after the 123815 other items are done.

Will save all the incarns, even if druids don't deserve it.

Hekili commented on TheWarWithin/DruidGuardian.lua:
```diff
@@ -1963,13 +2006,17 @@ spec:RegisterAbilities( {
         talent = "skull_bash",
         startsCombat = true,
-
+        form = function () if talent.fluid_form.enabled then
```
🤮

Hekili commented on TheWarWithin/DruidRestoration.lua:
```diff
@@ -19,116 +19,137 @@ spec:RegisterResource( Enum.PowerType.Rage )
     improved_ironbark           = {  82081, 382552, 1 }, -- Ironbark's cooldown is reduced by 20 sec.
     improved_regrowth           = {  82055, 231032, 1 }, -- Regrowth's initial heal has a 40% increased chance for a critical effect if the target is already affected by Regrowth.
     improved_wild_growth        = {  82045, 328025, 1 }, -- Wild Growth heals 1 additional target.
-    incarnation                 = {  82064,  33891, 1 }, -- Shapeshift into the Tree of Life, increasing healing done by 10%, increasing armor by 120%, and granting protection from Polymorph effects. Functionality of Rejuvenation, Wild Growth, Regrowth, Entangling Roots, and Wrath is enhanced. Lasts 30 sec. You may shapeshift in and out of this form for its duration.
```
Save it.

syrifgit replied:
~~This might just be a merging / conflcit thing? it's there on my editor~~ oh nvm I see what's happening now.

Hekili commented on TheWarWithin/DruidRestoration.lua:
```diff
@@ -498,6 +523,19 @@ spec:RegisterHook( "runHandler", function( ability )
 
+-- The War Within
+spec:RegisterGear( "tww2", 229310, 229308, 229306, 229307, 229305  )
+spec:RegisterAuras( {
```
If it has no auras, it does not need the RegisterAuras call.

syrifgit replied:
Leftover copy/paste tier set template, the resto auras ended up being pretty irrelevant, same with all the other healers I think.

Hekili commented on TheWarWithin/DruidRestoration.lua:
```diff
@@ -930,6 +968,30 @@ spec:RegisterAbilities( {
         end,
     },
 
+    skull_bash = {
```
Is this Resto version different from the other specs?  It should've inherited from an early module.

syrifgit replied:
Hm, don't think so. I think I just copied it in for completeness. ctrl-f an ability and it not be in the spec file got annoying after 20 specs.

If you prefer to leave it minimal it doesn't need to be there, let me know. I also need to delete the one from Balance because they can't cast it anymore.

Hekili commented on TheWarWithin/DruidRestoration.lua:
```diff
@@ -930,6 +968,30 @@ spec:RegisterAbilities( {
+
+        talent = "skull_bash",
+        startsCombat = true,
+        form = function () if talent.fluid_form.enabled then
```
I am under attack.

Hekili commented on TheWarWithin/EvokerDevastation.lua:
```diff
@@ -710,71 +720,110 @@ spec:RegisterStateExpr( "maximum", function()
+    if ability.empowered then
+        if talent.power_swell.enabled then applyBuff( "power_swell" ) end -- TODO: Modify Essence regen rate.
+        if talent.animosity.enabled then
+            animosityExtension = animosityExtension + 1
```
These casts have to be counted in the combat log and addressed in `reset_precast` or else it will get incremented based on every set of recommendations being generated instead of tracked based on what happened for real in the game.

CLEU:  Move `local animosityExtension = 0` above CLEU handler.  When `animosity` is talented, watch `SPELL_CAST_SUCCESS` and reset counter to 0 when you cast Dragonrage? and increment when you cast a red or blue spell (unless you've hit the max extensions).

`spec:RegisterStateExpr( "animosity_extension", function() return animosityExtension end )`

reset_precast:  `animosity_extension = nil`

This makes it so that `animosityExtension` is always the real number of extensions; `animosity_extension` will pull from the real number until we modify it (`animosity_extension = animosity_extension + 1`) and then it gets reset in `reset_precast` when the next set of recommendations is generated.

Time travel stuff.

syrifgit replied:
I think the thing I am having trouble understanding about this concept (and same with BM hunter) is: manipulating the variable (extension counter here, index number for BM) for use throughout consecutive recommendations in a set, while maintaining the real game-state value at the beginning of each set. The part that makes it more confusing here is that I can't ask the game what the counter/index is, as the trigger is a one-time event in the CLEU.

Lets say recommendations `#1` and `#2` of our set are both empower spells which will trigger the extension, and `#3` is some spell whose APL conditions take into account `buff.dragonrage.remains`. The previous cast was dragonrage, so we are fresh at `animosityExtension = 0`

1) Which part of this solution allows `#2` to see that there has already been 1 (virtualized / time travelled into the future) extension, and compare that value to 4 to confirm its allowed to extend again? 
2) Which part of it keeps the real counter accurate despite the above happening?

`animosityExtension = animosityExtension + 1` is this changing the real or virtual counter? I seem to be thinking that if we are manipulating, we would actually want `animosity_extension` to return a different variable, `animosityExtensionVirtual`. This variable would be synced during `reset_precast`via something like `animosityExtensionVirtual = animosityExtension`. This is the variable we would actually manipulate in ability handlers and the `run_handler`?

Is this the "easy" solution? Create a 2nd variable which you can manipulate with no fear, because it resets itself to your real value which is maintained by CLEU?

Hekili replied:
Yes, I think.  We keep one variable to represent actual data; in this case `animosityExtension` (camel case is usually internal, this_style is typically used for expressions that might appear in an APL).

> `animosityExtension = animosityExtension + 1` is this changing the real or virtual counter?

That would change the real counter; you asked this question because I messed up in my original comment (now corrected).

* In CLEU, we modify `animosityExtension`.
* In reset_precast and all handlers/references in our abilities, we use `animosity_extension`.

If it helps, you could rename the internal CLEU variable to `THIS_IS_REAL_ANIMOSITY` and then we know `animosity_extension` is the value that gets manipulated within a recommendation set.

More detail below, in case the above doesn't clear it up.

...

We need a separate variable that initially matches `animosityExtension`'s value when resetting (via reset_precast)., but can be manipulated over the course of making several recommendations.  

1. Keep `animosityExtension` sane; it is only manipulated by the appropriate *real* combat log or other events.  That means accounting for the event(s) that initialize it to 0 and the event(s) that increment it.

2. Create an expression that acts as an interface to `animosityExtension`; it accesses that value if needed, but cannot change its value.

```Lua
spec:RegisterStateExpr( "animosity_extension", function() return animosityExtension end )
```

This makes it so that `animosity_extension`, when not initialized, will return the *real* value (without manipulating `animosityExtension` itself).

3. In `reset_precast`, we wipe `animosity_extension` so that for this set of recommendations, it starts as uninitialized (returning the real value) until it gets modified in a handler or other function again.

`animosityExtension` is always real.  `animosity_extension` references the real value until we manipulate it, so that it is durable through this set of recommendations.

Combat Log:  `animosityExtension` gets incremented from 0 to 1 by the appropriate event.

Recommendation Set
1. reset_precast: `animosity_extension = nil`
  * `animosity_extension` will reference `animosityExtension`'s value.
  * When `state.animosity_extension` is nil, we get the value of `animosityExtension` instead.
2.  Recommendation 1:  The selected ability extends animosity with `animosity_extension = animosity_extension + 1`.
  * `animosity_extension` is still nil, so `animosityExtension` is automatically referenced in its place.
  * This is essentially `animosity_extension = [ value of animosityExtension = 1 ] + 1`.
  * `animosity_extension` is now set to 2 for the remainder of this set of recommendations, until/unless it is incremented or set to 0 by another handler in a subsequent ability.; it will *not* reference `animosityExtension` any longer, until/unless `animosity_extension` is set to `nil` again.
  * It will get set to `nil` again when `reset_precast` fires at the start of the next display's recommendations, so the first use of `animosity_extension` in that set of recommendations will pull from `animosityExtension` again.

Hekili commented on TheWarWithin/EvokerDevastation.lua:
```diff
@@ -710,71 +720,110 @@ spec:RegisterStateExpr( "maximum", function()
+        if talent.power_swell.enabled then applyBuff( "power_swell" ) end -- TODO: Modify Essence regen rate.
+        if talent.animosity.enabled then
+            animosityExtension = animosityExtension + 1
+            if animosityExtension < 5 then
```
Do we get 4 or 5 extensions from Animosity?  We could probably just combine these animosity-related `if` statements:

```Lua
if talent.animosity.enabled and animosity_extension < 5 then -- < 4?
    animosity_extension = animosity_extension + 1
    buff.dragonrage.expires = buff.dragonrage.expires + 5
end
```

syrifgit replied:
Yep that is the same. You get 4 extensions of 5 seconds each for a max of 20. Will convert to remove nested if.

Hekili commented on TheWarWithin/EvokerPreservation.lua:
```diff
@@ -116,19 +129,19 @@ spec:RegisterTalents( {
     resonating_sphere               = {  93258, 376236, 1 }, -- Temporal Anomaly applies Echo at 30% effectiveness to the first 5 allies it passes through.
-    reversion                       = {  93338, 366155, 1 }, -- Reverse an ally's injuries, instantly healing them for 15% of damage taken in the last 5 sec and an additional 30,334 over 12 sec. When Reversion critically heals, its duration is extended by 2 sec, up to a maximum of 12 sec.
-    rewind                          = {  93337, 363534, 1 }, -- Rewind 33% of damage taken in the last 5 seconds by all allies within 40 yds. Always heals for at least 20,994. Healing increased by 100% when not in a raid.
+    reversion                       = {  93338, 366155, 1 }, -- Reverse an ally's injuries, instantly healing them for 15% of damage taken in the last 5 sec and an additional 58,287 over 15.6 sec. When Reversion critically heals, its duration is extended by 2 sec, up to a maximum of 12 sec.
```
Just a quick note; it's good to be naked when using the skeleton generator, so that the durations aren't impacted by haste on gear.

I don't know if that's the case here, or if Reversion naturally lasts 15.6 seconds.  Not a bug, no fix needed -- just info.

Hekili commented on TheWarWithin/HunterBeastMastery.lua:
```diff
@@ -778,11 +775,33 @@ spec:RegisterAuras( {
+        duration = 20,
+        max_stack = 4,
+    },
+    -- Keep an eye out for what SIMC labels all these auras, be ready to change the names
```
sc_hunter.cpp looks to be already updated.

```C++
  buffs.howl_of_the_pack_leader_wyvern_ready = 
    make_buff( this, "howl_of_the_pack_leader_wyvern_ready", talents.howl_of_the_pack_leader_wyvern_ready_buff )
      ->set_stack_change_callback(
        [ this ]( buff_t* b, int, int cur ) {
          if ( cur == 0 && !buffs.howl_of_the_pack_leader_cooldown->check() )
            buffs.howl_of_the_pack_leader_cooldown->trigger();
        } );

  buffs.howl_of_the_pack_leader_boar_ready = 
    make_buff( this, "howl_of_the_pack_leader_boar_ready", talents.howl_of_the_pack_leader_boar_ready_buff )
      ->set_stack_change_callback(
        [ this ]( buff_t* b, int, int cur ) {
          if ( cur == 0 && !buffs.howl_of_the_pack_leader_cooldown->check() )
            buffs.howl_of_the_pack_leader_cooldown->trigger();
        } );

  buffs.howl_of_the_pack_leader_bear_ready = 
    make_buff( this, "howl_of_the_pack_leader_bear_ready", talents.howl_of_the_pack_leader_bear_ready_buff )
      ->set_stack_change_callback(
        [ this ]( buff_t* b, int, int cur ) {
          if ( cur == 0 && !buffs.howl_of_the_pack_leader_cooldown->check() )
            buffs.howl_of_the_pack_leader_cooldown->trigger();
        } );

  buffs.howl_of_the_pack_leader_cooldown = 
    make_buff( this, "howl_of_the_pack_leader_cooldown", talents.howl_of_the_pack_leader_cooldown_buff )
      ->apply_affecting_aura( talents.better_together )
      ->set_stack_change_callback(
        [ this ]( buff_t* b, int, int cur ) {
          if ( cur == 0 )
            trigger_howl_of_the_pack_leader_ready();
        } );
```

syrifgit replied:
I had pretty good guesses!

Hekili commented on TheWarWithin/HunterBeastMastery.lua:
```diff
@@ -1212,6 +1229,40 @@ spec:RegisterAuras( {
+spec:RegisterGear( "tww2", 229271, 229272, 229274, 229270, 229273 )
+spec:RegisterAuras( {
+   -- 2-set
+    jackpot = {
```
Yeah, this is apparently a fake buff in SimC as well.  No change needed.

https://github.com/simulationcraft/simc/blob/thewarwithin/engine/class_modules/sc_hunter.cpp#L9012

syrifgit replied:
In practice, it's an instant-effect that just sorta happens and does a thing. Not predictable outside of casting your major cooldown (for most ranged specs).

Hekili commented on TheWarWithin/HunterBeastMastery.lua:
```diff
@@ -1245,8 +1295,56 @@ end )
+    "howl_of_the_pack_leader_bear",
+}
+
+local pack_leader_buff_current = 1
```
This needs to be a RegisterSpecExpr and have it get initialized from the value you're tracking in CLEU.  And reset this to `nil` on reset.  (Like the other thing I just noted.)

Hekili commented on TheWarWithin/HunterBeastMastery.lua:
```diff
@@ -1294,26 +1391,39 @@ for i = 1, 40 do
 end
 
 spec:RegisterHook( "COMBAT_LOG_EVENT_UNFILTERED", function( _, subtype, _, sourceGUID, sourceName, _, _, destGUID, destName, destFlags, _, spellID, spellName )
```
```Lua
local buffLeaderPackCurrent = 0
```

```Lua
spec:RegisterSpecFunction( "buff_leader_pack_current", function() return buffLeaderPackCurrent end )

-- in reset_precast:
buff_leader_pack_current = nil
```

Hekili replied:
(Or `pack_leader_buff_current`  I didn't intentionally change the words around.)

Hekili commented on TheWarWithin/HunterBeastMastery.lua:
```diff
@@ -1294,26 +1391,39 @@ for i = 1, 40 do
+            for index, buffName in ipairs( pack_leader__buff_cycle ) do
+                if spellID == spec.auras[ buffName ].id then
+                    -- Align the cycle to the buff AFTER the currently applied one.
+                    pack_leader_buff_current = ( index % #pack_leader__buff_cycle ) + 1
```
Update the camel-case one here.  Make sure the table declaration for `pack_leader__buff_cycle` is located before the combat log handler in this file.

Hekili commented on TheWarWithin/HunterBeastMastery.lua:
```diff
@@ -1899,6 +1996,17 @@ spec:RegisterAbilities( {
+        gcd = "off",
+        hidden = true,
+        handler = function ()
+
```
Does nothing?

syrifgit replied:
Yeah, wasn't sure if I needed to fill it in or not. Answer is no, will remove.

Hekili commented on TheWarWithin/PaladinHoly.lua:
```diff
@@ -588,8 +588,27 @@ spec:RegisterAuras( {
+spec:RegisterGear( "tww2", 229244, 229242, 229243, 229245, 229247 )
+spec:RegisterAuras( {
+   -- 2-set
+    set_placeholder_2 = {
```
If it's this ID, it will be `winning_streak`.

syrifgit replied:
This is just the copy/paste from Ret pal, wasn't sure at the time if it would be filled in or not. I will remove both auras here, as with other healer auras for this tier they are largely useless and unlikely to ever be referenced by even the largest of nerds.

Hekili commented on TheWarWithin/PaladinHoly.lua:
```diff
@@ -588,8 +588,27 @@ spec:RegisterAuras( {
+        duration = 30,
+        max_stack = 10
+    },
+    set_placeholder_4 = {
```
`all_in` ?

Hekili commented on TheWarWithin/PaladinProtection.lua:
```diff
@@ -798,6 +798,20 @@ spec:RegisterAuras( {
+    -- https://www.wowhead.com/ptr-2/spell=1218114/luck-of-the-draw
+    -- Each time you take damage you have a chance to activate Luck of the Draw! causing you to cast Guardian of Ancient Kings for 4.0 sec. Your damage done is increased by 15% for 10 sec after Luck of the Draw! activates.
+    luck_of_the_draw = {
+    id = 1218114,
```
If there's only one, it can be `RegisterAura` without the extra table wrapper.  And the indentation can be fixed.  :D

Hekili commented on TheWarWithin/ShamanElemental.lua:
```diff
@@ -18,163 +18,176 @@ spec:RegisterResource( Enum.PowerType.Mana )
     ancestral_swiftness         = {  94894, 443454, 1 }, -- Your next healing or damaging spell is instant, costs no mana, and deals 10% increased damage and healing. If you know Nature's Swiftness, it is replaced by Ancestral Swiftness and causes Ancestral Swiftness to call an Ancestor to your side for 6 sec.
     ancient_fellowship          = {  94862, 443423, 1 }, -- Ancestors have a 20% chance to call another Ancestor for 6 sec when they depart.
-    call_of_the_ancestors       = {  94888, 443450, 1 }, -- Primordial Wave calls an Ancestor to your side for 6 sec. Whenever you cast a healing or damaging spell, the Ancestor will cast a similar spell.
+    call_of_the_ancestors       = {  94888, 443450, 1, "farseer" }, -- Primordial Wave calls an Ancestor to your side for 6 sec. Whenever you cast a healing or damaging spell, the Ancestor will cast a similar spell.
```
Oof, that's an 11.0.7 bug too, then.

syrifgit replied:
Yeah :/ It got lost in the sauce on October 23

Hekili commented on TheWarWithin/ShamanElemental.lua:
```diff
@@ -209,21 +222,21 @@ spec:RegisterAuras( {
     -- https://wowhead.com/beta/spell=114050
     ascendance = {
-        id = 114050,
+        id = 1219480,
```
Ignore my eyeroll that they changed this buff/ID.  🙄

syrifgit replied:
And didn't change the other 2 specs

Hekili commented on TheWarWithin/ShamanElemental.lua:
```diff
@@ -1452,12 +1472,11 @@ spec:RegisterAbilities( {
-                    state:QueueAuraEvent( "ascendance", TriggerStaticAccumulation, query_time + i, "AURA_PERIODIC" )
-                end
-            end
+            spec.abilities.flame_shock.handler()
```
Based on the above, I'm assuming that Ascendance gets its own version in Enhancement and Resto files?  If so, we don't need to have `id` as a function nor the copy line with the other `id`s.

syrifgit replied:
They do, will change

Hekili commented on TheWarWithin/ShamanElemental.lua:
```diff
@@ -2753,22 +2780,20 @@ spec:RegisterAbilities( {
 
         velocity = 30,
```
Does Primordial Wave have a velocity on PTR?  It looks like it doesn't from the spell data.

syrifgit replied:
It's a projectile with a travel time if that means yes? But they changed it so that pretty much all interactions happen on cast because everyone hated the on-impact.

Hekili commented on TheWarWithin/ShamanEnhancement.lua:
```diff
@@ -1909,7 +1954,10 @@ spec:RegisterAbilities( {
-            if buff.whirling_fire.up then removeBuff( "whirling_fire" ) end
+            if buff.whirling_earth.up then
+                removeBuff( "whirling_earth" )
+                active_dot.flame_shock = min( active_enemies, active_dot.flame_shock + 5 )
```
This should probably be `true_active_enemies` but it also shouldn't matter.  :D

syrifgit replied:
`spreadActiveDots()` always uses `true_active_enemies` ;)

Hekili commented on TheWarWithin/ShamanEnhancement.lua:
```diff
@@ -2404,7 +2455,29 @@ spec:RegisterAbilities( {
+            consume_maelstrom()
+        end,
+
+    },
```
This will need a `bind = "primordial_wave",` and vice versa for the keybind detection.  And probably a `flash = `*pwave_spell_id*`,` for SpellFlash.

Hekili commented on TheWarWithin/ShamanEnhancement.lua:
```diff
@@ -2446,8 +2521,7 @@ spec:RegisterAbilities( {
         end,
 
         impact = function ()
```
If everything happens on cast, no `impact` needed and `velocity` should be removed (avoids wasting time looking for an impact even that won't happen).  Also, I already don't remember -- does Ele have its own copy of PWave in its file?  If so, we don't need to have the Elemental vs. Enhancement check for Primal Maelstrom.

syrifgit replied:
`Also, I already don't remember` 5 mins is a long time since you commented on ele's Pwave. ;) Will remove spec-check stuff.

Hekili commented on TheWarWithin/ShamanEnhancement.lua:
```diff
@@ -2638,7 +2712,11 @@ spec:RegisterAbilities( {
         handler = function ()
             if not talent.molten_thunder.enabled then applyDebuff( "target", "sundering" ) end
-
+            -- Todo: Track 11.1 molten thunder reset and whether or not it can be 100%. Currently a serverside script so *shrug*
```
Tell me more.  There may be a hidden aura that we can actually track.

syrifgit replied:
I tried some stuff in-game. PTR Page: https://www.wowhead.com/ptr-2/spell=469344/molten-thunder

Effect #2 and #3 represent the 40% and scaling 12% * enemy chance to reset. 

Could be more stuff I haven't tried though maybe?

Hekili commented on TheWarWithin/WarriorFury.lua:
```diff
@@ -539,16 +539,40 @@ spec:RegisterAuras( {
+    --https://www.wowhead.com/ptr-2/spell=1216569/double-down
+    --Raging Blow damage increased by 15% and Rage generated increased by (w2 / 10).
+    double_down_rb = {
+    id = 1216569,
```
Indent.

Hekili commented on TheWarWithin/WarriorFury.lua:
```diff
@@ -706,6 +724,11 @@ spec:RegisterStateExpr( "glory_rage", function ()
     return gloryRage
 end )
 
+spec:RegisterHook( "prespend", function( amt, resource, overcap, clean )
```
Ok, I think I get it now, this is making Recklessness double the *generation* of Rage.  This probably needs an `and amt < 0` added so that we don't also double the *cost* of Rage for spenders.

Hekili commented on TheWarWithin/WarriorFury.lua:
```diff
@@ -949,14 +972,15 @@ spec:RegisterAbilities( {
         startsCombat = true,
         texture = 236304,
-        buff = "reckless_abandon_bloodbath",
+        buff = "reckless_abandon",
```
Is this a change to Reckless Abandon?  Because I think I needed to distinguish the effect with a pseudo-buffs, but if that's not needed anymore, that's great.

syrifgit replied:
Correct, reworked reckless abandon is much easier for us.

Hekili commented on TheWarWithin/WarriorFury.lua:
```diff
@@ -972,35 +996,35 @@ spec:RegisterAbilities( {
                 addStack( "cadence_of_fujieda" )
             end
-        
-            if buff.reckless_abandon_bloodbath.up then removeBuff( "reckless_abandon_bloodbath" ) end
```
It doesn't remove Reckless Abandon?

syrifgit replied:
it's not a one-time use anymore. It's transformed for the whole duration of recklessness.

```
Reckless Abandon has been redesigned – Activating Recklessness generates 50 Rage and while Recklessness is active, Raging Blow and Bloodthirst are replaced by Crushing Blow and Bloodbath.
Crushing Blow – A powerful dual-weapon strike that deals 10% increased Critical Strike damage.
Bloodbath – A powerful attack that applies a Bleed to the target that is extended by its full duration when reapplied.
Developers’ notes: Reckless Abandon has been having an outsized impact on rotations and we’d like to bring the focus of the talent back to Recklessness as well as make Crushing Blow and Bloodbath abilities with unique additional effects.
```

Hekili commented on TheWarWithin/WarriorFury.lua:
```diff
@@ -1882,7 +1907,7 @@ spec:RegisterAbilities( {
         end,
 
         handler = function ()
-            if talent.improved_whirlwind.enabled or talent.meat_cleaver.enabled then
```
Not sure why the Meat Cleaver check was there; is this ability shared with Arms?  This is probably fine.

syrifgit replied:
I think I removed this already on some other W-I-P branch for warrior. Will remove here too, redundant. The meat_cleaver buff and whirlwind buff are already copied together anyways.

improved WW is a mandatory pre-req talent for meat cleaver right now too. Double redundant.

syrifgit replied:
.... some of these comments are popping up on my main branch and it's very confusing in my editor lmao. I already deleted this on the actual PR I think.

Hekili commented on TheWarWithin/WarlockAffliction.lua:
```diff
@@ -868,23 +880,40 @@ spec:RegisterHook( "COMBAT_LOG_EVENT_UNFILTERED", function( _, subtype, _, sourc
+        max_stack = 1
+    },
+-- https://www.wowhead.com/ptr-2/spell=1219036/warlock-affliction-11-1-class-set-4pc
+    tww2_set_haste_buff = {
```
I don't think we need this; we just anticipate its effects in any handler by checking for `set_bonus.tww2 > 3` or whatever when we apply/spend Jackpot!

syrifgit replied:
Wasn't sure, some of the stuff like this is just more "wait to see what the APLs use" because I'd rather delete it later than go back and find it to re-add.

Though a haste buff is pretty unlikely to be used.

Hekili commented on TheWarWithin/PaladinRetribution.lua:
```diff
@@ -404,23 +396,9 @@ spec:RegisterAuras( {
         id = 198034,
-        duration = 12,
-        max_stack = 1,
-        generate = function( dh )
```
Are you sure we don't need it like this?  Does it actually apply the buff (visibly) now?

syrifgit replied:
![image](https://github.com/user-attachments/assets/056a9235-5e13-4f1a-9a38-4902ba5cd9d1)

Hekili commented on TheWarWithin/PaladinRetribution.lua:
```diff
@@ -1618,6 +1591,17 @@ spec:RegisterAbilities( {
+
+            -- Hero Talents
+            if talent.lights_deliverance.enabled and talent.hammerfall.enabled then
+                if buff.shake_the_heavens.up then
```
`addStack( "lights_deliverance, nil, 1 + ( buff.shake_the_heavens.up and 1 or 0 ) )`

syrifgit replied:
neat

Hekili commented on TheWarWithin/PaladinRetribution.lua:
```diff
@@ -1798,24 +1780,33 @@ spec:RegisterAbilities( {
     -- Hammer down your enemy with the power of the Light, dealing $429826s1 Holy damage and ${$429826s1/2} Holy damage up to 4 nearby enemies. ; Additionally, calls down Empyrean Hammers from the sky to strike $427445s2 nearby enemies for $431398s1 Holy damage each.;
     hammer_of_light = {
```
Ok, so we are giving each their own version of Hammer of Light.  That's fine.

syrifgit replied:
Yes pls. IMO if you need to start doing `state.spec` checks in multiple parts of the ability registration, they are different enough to have their own versions. Gets hard to wrap your head around for changes sometimes. The skull bash from druids was one where it literally was identical.

Hekili commented on TheWarWithin/PaladinRetribution.lua:
```diff
@@ -2052,7 +2038,7 @@ spec:RegisterAbilities( {
             end
 
-            if azerite.empyreal_ward.enabled then applyBuff( "empyreal_ward" ) end
+            if talent.empyreal_ward.enabled then applyBuff( "empyreal_ward" ) end
```
Technically, both?  Also, didn't this aura get removed in your edits above?

syrifgit replied:
I figured since the azerite effect has been replaced by a talent which does the same thing (with a different aura ID), keeping both versions was more confusing than it was worth for BFA support. Can revert to support both if you like!

Hekili replied:
It's probably not super important, except minor impact in Timewalking.  We can leave it.

Hekili commented on TheWarWithin/PaladinRetribution.lua:
```diff
@@ -2239,38 +2223,43 @@ spec:RegisterAbilities( {
-                removeBuff( "empyrean_legacy" )
+            -- Hero Talents
+            if talent.lights_deliverance.enabled and talent.hammerfall.enabled then
+                if buff.shake_the_heavens.up then
```
Same as other comment.

Hekili commented on TheWarWithin/PriestHoly.lua:
```diff
@@ -229,8 +226,8 @@ spec:RegisterAuras( {
         id = 64843,
-        duration = 8,
-        tick_time = 2,
+        function() return 5 * haste end,
```
Missing `duration =` ?

syrifgit replied:
Interesting that I managed to not trigger a LUA error on that somehow.

Hekili commented on TheWarWithin/PriestHoly.lua:
```diff
@@ -229,8 +226,8 @@ spec:RegisterAuras( {
-        duration = 8,
-        tick_time = 2,
+        function() return 5 * haste end,
+        tick_time = function() return ( 5 * haste ) / 5 end,
```
Reduce your math!

`return haste`

syrifgit replied:
atleast I didn't do math on a boolean. We won't talk about this one. 

Don't look at resto druid Tranq either (which is done identically) thanks.

Hekili commented on TheWarWithin/PriestHoly.lua:
```diff
@@ -436,7 +438,12 @@ spec:RegisterAuras( {
     },
 } )
 
+local naaruMulti = 1 + ( 0.1 * state.talent.light_of_the_naaru.rank )
```
I haven't looked, but this needs to be updated in reset_precast or it will be minorly wrong upon changing talents.

syrifgit replied:
Ah, I had assumed a talent change would reroll through this section so I made it static, will add.

Hekili commented on TheWarWithin/MonkWindwalker.lua:
```diff
@@ -101,124 +116,105 @@ spec:RegisterTalents( {
     inner_peace                    = { 101214, 397768, 1 }, -- Increases maximum Energy by 30. Tiger Palm's Energy cost reduced by 5.
-    invoke_xuen                    = { 101206, 123904, 1 }, -- Summons an effigy of Xuen, the White Tiger for 20 sec. Xuen attacks your primary target, and strikes 3 enemies within 10 yards every 0.9 sec with Tiger Lightning for 10,817 Nature damage. Every 4 sec, Xuen strikes your enemies with Empowered Tiger Lightning dealing 8% of the damage you have dealt to those targets in the last 4 sec.
-    invoke_xuen_the_white_tiger    = { 101206, 123904, 1 }, -- Summons an effigy of Xuen, the White Tiger for 20 sec. Xuen attacks your primary target, and strikes 3 enemies within 10 yards every 0.9 sec with Tiger Lightning for 10,817 Nature damage. Every 4 sec, Xuen strikes your enemies with Empowered Tiger Lightning dealing 8% of the damage you have dealt to those targets in the last 4 sec.
+    invoke_xuen_the_white_tiger    = { 101206, 123904, 1 }, -- Summons an effigy of Xuen, the White Tiger for 20 sec. Xuen attacks your primary target, and strikes 3 enemies within 10 yards every 0.9 sec with Tiger Lightning for 3,870 Nature damage. Every 4 sec, Xuen strikes your enemies with Empowered Tiger Lightning dealing 8% of the damage you have dealt to those targets in the last 4 sec.
```
Oh, this (and all Monks) need their abbreviated copy of their `invoke_x` talents shortened bc it's used that way throughout the code.

I.e., need a copy of this that is just `invoke_xuen`.

Hekili commented on TheWarWithin/MonkWindwalker.lua:
```diff
@@ -101,124 +116,105 @@ spec:RegisterTalents( {
     hardened_soles                 = { 101047, 391383, 1 }, -- Blackout Kick critical strike chance increased by 10% and critical damage increased by 18%.
     hit_combo                      = { 101216, 196740, 1 }, -- Each successive attack that triggers Combo Strikes in a row grants 1% increased damage, stacking up to 5 times.
     inner_peace                    = { 101214, 397768, 1 }, -- Increases maximum Energy by 30. Tiger Palm's Energy cost reduced by 5.
-    invoke_xuen                    = { 101206, 123904, 1 }, -- Summons an effigy of Xuen, the White Tiger for 20 sec. Xuen attacks your primary target, and strikes 3 enemies within 10 yards every 0.9 sec with Tiger Lightning for 10,817 Nature damage. Every 4 sec, Xuen strikes your enemies with Empowered Tiger Lightning dealing 8% of the damage you have dealt to those targets in the last 4 sec.
```
This line is sad to be deleted.

Hekili commented on TheWarWithin/MonkWindwalker.lua:
```diff
@@ -965,6 +987,12 @@ spec:RegisterHook( "reset_precast", function ()
     chiSpent = 0
 
+    if talent.ordered_elements.enabled then
+        if buff.storm_earth_and_fire.up then orderedElementsMod = 1
```
Careful; this will make the `orderedElementsMod` remain at this value for the entire prediction set even if SEF only has 0.01s remaining.  You'd have to update this value on application/expiration of Storm, Earth, and Fire or you can use a state expression to replace it.  That doesn't give you the efficiency that I think you were looking for, though.

```Lua
spec:RegisterStateExpr( "orderedElementsMod", function()
    if not talented.ordered_elements.enabled or buff.storm_earth_and_fire.down then return 0 end
    return 1
end )
```

syrifgit replied:
I will use state expression for now to get it fully functional. That's what this iteration was supposed to be lol. I still intend to (eventually) come up with a more all-inclusive solution for this concept of a specwide resource discount under certain conditions (mana tea, ordered elements, etc).

Hekili commented on TheWarWithin/MonkWindwalker.lua:
```diff
@@ -999,10 +1027,10 @@ end )
 spec:RegisterStateTable( "spinning_crane_kick", setmetatable( { onReset = function( self ) self.count = nil end },
         { __index = function( t, k )
-            if k == "count" then
+            --[[if k == "count" then
```
Why is this removed?  `spinning_crane_kick.count` has (had?) a lot of reference in code and APLs.

syrifgit replied:
mark of the crane is boom, so nothing to return anymore.

![image](https://github.com/user-attachments/assets/d3835330-5ffd-45db-8190-b50fe7356a84)

syrifgit committed:
big 'ole review sesh

Remaining items from review

Dev Evoker dragonrage extension
BM (And surv) pack leader cycling buff
WW Chi cost

syrifgit committed:
Time travel or something

Are variables real or fake?

The Ship of Theseus, also known as Theseus's Paradox, is a paradox and a common thought experiment about whether an object is the same object after having all of its original components replaced over time, typically one after the other.

syrifgit committed:
Update HunterSurvival.lua

syrifgit committed:
WW accuracy stuff

syrifgit committed:
Jan 28 + Feb 6

syrifgit commented:
Now includes everything except for the datamining that just came out. Might be a few things to tweak, will wait for the written notes.

syrifgit committed:
February 12th patch notes
